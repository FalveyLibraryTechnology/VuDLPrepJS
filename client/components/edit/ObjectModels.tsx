import React from "react";
import { useEditorContext } from "../../context/EditorContext";

export interface ObjectModelsProps {
    pid: string;
}

const ObjectModels = ({ pid }: ObjectModelsProps): React.ReactElement => {
    const {
        state: { objectDetailsStorage },
    } = useEditorContext();
    const models = objectDetailsStorage?.[pid]?.models ?? [];
    return models.length > 0 ? (
        <div>
            <b>Models:</b>
            <ul>
                {models.sort().map((model: string) => {
                    const modelName = model.replace("vudl-system:", "");
                    return (
                        <li style={{ border: 0 }} key={modelName}>
                            {modelName}
                        </li>
                    );
                })}
            </ul>
        </div>
    ) : (
        <div></div>
    );
};

export default ObjectModels;
