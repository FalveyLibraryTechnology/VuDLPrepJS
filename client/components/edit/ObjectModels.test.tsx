import React from "react";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";
import renderer from "react-test-renderer";
import ObjectModels from "./ObjectModels";

const mockUseEditorContext = jest.fn();
jest.mock("../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

describe("ObjectModels", () => {
    let editorValues;
    beforeEach(() => {
        editorValues = {
            state: {
                objectDetailsStorage: {},
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });

    it("renders models", async () => {
        editorValues.state.objectDetailsStorage["foo:123"] = {
            models: ["vudl-system:CoreModel", "vudl-system:CollectionModel", "vudl-system:FolderCollection"],
        };
        const tree = renderer.create(<ObjectModels pid="foo:123" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders nothing when data is missing", async () => {
        const tree = renderer.create(<ObjectModels pid="foo:123" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
