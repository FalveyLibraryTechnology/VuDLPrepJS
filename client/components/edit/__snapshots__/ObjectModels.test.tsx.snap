// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectModels renders models 1`] = `
<div>
  <b>
    Models:
  </b>
  <ul>
    <li
      style={
        {
          "border": 0,
        }
      }
    >
      CollectionModel
    </li>
    <li
      style={
        {
          "border": 0,
        }
      }
    >
      CoreModel
    </li>
    <li
      style={
        {
          "border": 0,
        }
      }
    >
      FolderCollection
    </li>
  </ul>
</div>
`;

exports[`ObjectModels renders nothing when data is missing 1`] = `<div />`;
